### ISSUE TITLE
Analytics: Site Kit created properties do not appear in dropdowns

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#405]
## Bug Description

When configuring Analytics, the user has the option to create a new property within an existing account for the current site. When choosing this option, the property is created, and setup continues successfully.

However, when navigating to the settings for Analytics, it displays the current account, property and view IDs, but when editing, the dropdowns for Property and Profile/view are empty because they do not contain the current values.

This appears to be due to the cached `analytics::get-accounts` data in session storage containing the result from that request from before the new property was added. Once this is cleared, the dropdowns display correctly.

## Steps to reproduce

1. Setup Site Kit
1. Set up Analytics
    * Choose **Set Up New Property**
1. Go to Settings
1. Edit the settings for Analytics
1. See that only the Account choice is selected in the dropdowns; Property and View are blank without the new property choices available.

## Screenshots

![image](https://user-images.githubusercontent.com/1621608/63084024-e616fb00-bf52-11e9-8b25-2e7a029f5d1c.png)

## Additional Context

 - PHP Version: 7.2
 - OS: MacOS
 - Browser: Chrome
 - Plugin Version: 1.0.4

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## Changelog entry

*

**Comment by grzegorz-janoszka** [comment]
Sitekit just chooses the first property and it is not possible to choose anything else from the grayed dropdown.

**Comment by aaemnnosttv** [comment]
@grzegorz-janoszka the dropdowns are disabled in some cases based on the presence of an existing analytics tag on the site so it's likely that you're seeing the expected behavior. This issue is not related to what you're describing though. 

Please open a new issue for assistance with your specific case.

**Comment by felixarntz** [comment]
IB ✅ 

One minor thing: @aaemnnosttv The cache should be cleared using the new `data.invalidateCacheGroup()` function from #406 / #436. This issue's implementation is blocked by that anyway, so it would be great if you could just update this part of the IB. Other than that, looks great.

**Comment by adamsilverstein** [comment]
Generally this works better than before, however I still some room for improvement. @felixarntz  can you please review and let me know what deserves follow up issues.

I tested this using the following steps:

* create a new site on poopy.life (http://poopy.life/create) - i got `http://ill-cobra.w5.poopy.life`
* install build from develop
* Setup Site Kit
* Set up Analytics
* Choose "Set Up New Property"

After choosing "Setup new property" during the Analytics setup flow or when changing Analytics settings, the new property appeared in the dropdown immediately.

After selecting "Set Up New Property" I had to push the button labeled "Configure Analytics".  Should that button text change to 'Create' or 'Create Property' so its clear what action you are taking?

After clicking the UI paused with no visible feedback - _a spinner or some indication of saving feels missing here._  

After a few seconds I very briefly see the new site is added to the property selector and the page immediately refreshes to the Dashboard. Feels a little disjointed, a save confirmation then redirect might be fine.

![](https://cl.ly/64466d712923/Screen%252520Recording%2525202019-09-06%252520at%25252001.47%252520PM.gif)

When I went to the settings screen and looked at Analytics settings, everything looked right with the newly created property selected.

**Comment by adamsilverstein** [comment]
Moved to approval as the reported issue is resolved. Assigning to @felixarntz for review of possible follow up issues.
